srtcore/srt.h

```diff
@@ -995,6 +995,20 @@ SRT_API int64_t srt_time_now(void);
 
 SRT_API int64_t srt_connection_time(SRTSOCKET sock);
 
+enum SRT_SYNC_CLOCK_TYPE
+{
+    SRT_SYNC_CLOCK_TYPE_STDCXX_STEADY      = 0, // C++11 std::chrono::steady_clock
+    SRT_SYNC_CLOCK_TYPE_GETTIME_MONOTONIC  = 1, // clock_gettime with CLOCK_MONOTONIC
+    SRT_SYNC_CLOCK_TYPE_WINQPC             = 2,
+    SRT_SYNC_CLOCK_TYPE_MACH_ABSTIME       = 3,
+    SRT_SYNC_CLOCK_TYPE_POSIX_GETTIMEOFDAY = 4,
+    SRT_SYNC_CLOCK_TYPE_AMD64_RDTSC        = 5,
```

Comment:
To be honest I think 'rdtsc' clock type should be eradicated from this codebase. I know that that is not the point of this PR so this is just a thought; it is almost as though the RDTSC values should be marked as 'depricated' as soon as they are publically exposed.

Reply:
This is just packing the old code from UDT into sensible frames.

Reply:
@rndi Do you mean `srt::sync::rdtsc()` function or the name of the enum value `..AMD64_RDTSC`?
The name was chosen because the effective time source for that case is:
```c++
asm volatile("rdtsc" : "=a"(lval), "=d"(hval));
```